lib/svg4everybody.js

```diff
@@ -0,0 +1,127 @@
+/*! svg4everybody v2.0.0 | github.com/jonathantneal/svg4everybody */
+
+function embed(svg, g) {
+	if (g) {
+		var viewBox = !svg.getAttribute('viewBox') && g.getAttribute('viewBox');
+		var fragment = document.createDocumentFragment();
+		var clone = g.cloneNode(true);
+
+		if (viewBox) {
+			svg.setAttribute('viewBox', viewBox);
+		}
+
+		while (clone.childNodes.length) {
+			fragment.appendChild(clone.firstChild);
+		}
+
+		svg.appendChild(fragment);
+	}
+}
+
+function loadreadystatechange(xhr) {
+	xhr.onreadystatechange = function () {
+		if (xhr.readyState === 4) {
+			var x = document.createElement('x');
+
+			x.innerHTML = xhr.responseText;
+
+			xhr.s.splice(0).map(function (array) {
+				embed(array[0], x.querySelector('#' + array[1].replace(/(\W)/g, '\\$1')));
+			});
+		}
+	};
+
+	xhr.onreadystatechange();
+}
+
+function svg4everybody(opts) {
+	opts = opts || {};
+
+	var uses = document.getElementsByTagName('use');
+	var nosvg;
+
+	if (LEGACY_SUPPORT) {
+		var fallback = opts.fallback || function (src) {
+			return src.replace(/\?[^#]+/, '').replace('#', '.').replace(/^\./, '') + '.png' + (/\?[^#]+/.exec(src) || [''])[0];
+		};
+
+		nosvg = 'nosvg' in opts ? opts.nosvg : /\bMSIE [1-8]\b/.test(navigator.userAgent);
+
+		if (nosvg) {
+			document.createElement('svg');
+			document.createElement('use');
+		}
+	}
+
+	var polyfill = 'polyfill' in opts ? opts.polyfill : LEGACY_SUPPORT ? (
+		nosvg || /\bEdge\/12\b|\bMSIE [1-8]\b|\bTrident\/[567]\b|\bVersion\/7.0 Safari\b/.test(navigator.userAgent) || (navigator.userAgent.match(/AppleWebKit\/(\d+)/) || [])[1] < 537
+	) : (
+		/\bEdge\/12\b|\bTrident\/[567]\b|\bVersion\/7.0 Safari\b/.test(navigator.userAgent) || (navigator.userAgent.match(/AppleWebKit\/(\d+)/) || [])[1] < 537
+	);
+
+	var validate = opts.validate;
+	var requestAnimationFrame = window.requestAnimationFrame || setTimeout;
```

Comment:
If you name this `rAF` it's safer, and you can save the `window` as the context here will always be window as svg4everybody is only meant to be used in the browser.